garb/garb_recv_loop.cpp

```diff
@@ -96,7 +103,35 @@ RecvLoop::loop()
                     uuid_  = cc.uuid;
                     seqno_ = cc.seqno;
                     gcs_.request_state_transfer (config_.sst(),config_.donor());
-                    gcs_.join(gu::GTID(cc.uuid, cc.seqno), 0);
+                    if(config_.recv_script().empty()) {
+                        gcs_.join(gu::GTID(cc.uuid, cc.seqno), 0);
+                    } else {
+                        log_info << "Starting SST script";
+                        p.execute("rw", NULL);
+
+                        sst_err_log = std::thread([&](){
+                            FILE* err_pipe = p.err_pipe();
+                            const int out_len = 1024;
+                            char out_buf[out_len];
+                            char* p;
+                            while ((p = fgets(out_buf, out_len, err_pipe)) != NULL) {
+                                log_info << "[SST script] " << out_buf;
+                            }
+                            log_info << "SST script ended";
+                            sst_ended = true;
+                            gcs_.close();
+                        });
+
+                        sst_out_log = std::thread([&](){
+                            FILE* err_pipe = p.pipe();
```

Comment:
Ctrl+c, ctrl+v? Variable name is confusing. This is stdin/out pipe, not stderr pipe.

Reply:
No, we are piping the script's stdout and stderr to the logfile - not doing anything with stdin.